cxx/distributions/normal_test.cc

```diff
@@ -21,5 +21,57 @@ BOOST_AUTO_TEST_CASE(test_simple)
   BOOST_TEST(nd.logp_score() == -4.7494000141508543, tt::tolerance(1e-6));
 }
 
-// TODO(thomaswc): Add more tests, including checking numbers versus
-// github.com/probcomp/cgpm/src/primitives/normal.py.
+BOOST_AUTO_TEST_CASE(logp_before_incorporate)
+{
+  PRNG prng;
+  Normal nd(&prng);
+
+  BOOST_TEST(nd.logp(6.0) == -4.4357424552958129, tt::tolerance(1e-6));
+  BOOST_TEST(nd.logp_score() == 0.0, tt::tolerance(1e-6));
+
+  nd.incorporate(5.0);
+  nd.unincorporate(5.0);
+
+  BOOST_TEST(nd.logp(6.0) == -4.4357424552958129, tt::tolerance(1e-6));
+  BOOST_TEST(nd.logp_score() == 0.0, tt::tolerance(1e-6));
+}
+
+BOOST_AUTO_TEST_CASE(sample)
+{
+  PRNG prng;
+  Normal nd(&prng);
+
+  for (int i = 0; i < 1000; i++) {
+    nd.incorporate(42.0);
+  }
+
+  double s = nd.sample();
+
+  BOOST_TEST(s > 40.0);
+  BOOST_TEST(s < 44.0);
+}
+
+BOOST_AUTO_TEST_CASE(incorporate_raises_logp) {
+  PRNG prng;
+  Normal nd(&prng);
+
+  double old_lp = nd.logp(10.0);
+  for (int i = 0; i < 10; i++) {
```

Comment:
Could you use prefix increment instead of postfix? (And below.) go/cstyle#Preincrement_and_Predecrement I think we might as well stick to the Google style guide unless there's a reason not to. I'm working on a PR for general cleanup along those lines.

Reply:
Done.